src/phoenix/server/app.py

```diff
@@ -226,6 +230,25 @@ async def get_response(self, path: str, scope: Scope) -> Response:
         return response
 
 
+class RequestOriginHostnameValidator(BaseHTTPMiddleware):
+    def __init__(self, trusted_hostnames: List[str], *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+        self._trusted_hostnames = trusted_hostnames
+
+    async def dispatch(
+        self,
+        request: Request,
+        call_next: RequestResponseEndpoint,
+    ) -> Response:
+        headers = request.headers
+        for key in "origin", "referer":
+            if not (url := headers.get(key)):
```

Comment:
This looks like it is going to miss the headers if they are capitalized.

Reply:
addressed offline: Starlette headers `__getitem__` is case [insensitive](https://github.com/encode/starlette/blob/e11684013fe5ca084f5bd4e54830512a4dff9618/starlette/datastructures.py#L534-L535)